src/Fantomas/CodePrinter.fs

```diff
@@ -2591,7 +2591,14 @@ and genPat astContext pat =
     match pat with
     | PatOptionalVal(s) -> !- (sprintf "?%s" s)
     | PatAttrib(p, ats) -> genOnelinerAttributes astContext ats +> genPat astContext p
-    | PatOr(p1, p2) -> genPat astContext p1 +> sepNln -- "| " +> genPat astContext p2
+    | PatOr(p1, p2) ->
+        let sepNlnUnlessLastEventIsNewline ctx =
```

Comment:
Please move to Context.fs, place it around sepNln if possible.
This looks like a useful helper function that could be used in other scenarios as well.